### PR TITLE
BIDSification wakehealth dataset

### PR DESCRIPTION
**Pull request by alexfoias**
To test :

```
git checkout af/bidsification_scripts
python scripts/curate_wakehealth.py -d PATH_TO_WAKEHEALTH_INPUT_DATASET
```
PATH_TO_WAKEHEALTH_INPUT_DATASET should have the following contents:
Screen Shot 2021-06-02 at 11 12 48 AM

Fixes https://github.com/neuropoly/data-management/issues/80

### CODE REVIEW

File: scripts/curate_wakehealth.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    for ds in subdatasets:
+        path_dataset = os.path.join(root_data, ds)
+        contents_ds = os.listdir(path_dataset)
+        contents_ds.sort()
+        roi_no = 1
+        previous_sub_id_bids = ''
+
+        # Loop across contents of each subdirectory
+        for item in contents_ds:
+            sub_id_initial = item.split(' - ')[0]
```
Comment (mariehbourget) [inline scripts/curate_wakehealth.py:53]: We would like to keep only the first 5 digits. For some files in `20210308_extracts`, there is no dash, ex:
`17-268 94144536 - 2017-09-27 22.44.10_x40_z0_extract20210526114558-11.tif`, the sub-id should be `17268`.

File: scripts/curate_wakehealth.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            if sub_id_initial.find('_'):
+                sub_id_initial = sub_id_initial.split('_')[0]
+            sub_id_initial = sub_id_initial.replace(' ', '')
+            sub_id_initial = sub_id_initial.replace('-', '')
+            path_item_in = os.path.join(path_dataset, item)
+            counter_file_in = counter_file_in + 1
+            sub_id_bids = 'sub-' + sub_id_initial
+            sample_id_bids = 'sample- ' + sub_id_initial
```
Comment (mariehbourget) [inline scripts/curate_wakehealth.py:61]:
```suggestion
            sample_id_bids = 'sample-' + sub_id_initial
```

File: scripts/curate_wakehealth.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            previous_sub_id_bids = sub_id_bids
+
+            # Curate preview
+            if item.endswith('_x0.625_z0.tif'):
+                item_out = sub_id_bids + '_' + sample_id_bids + '_photo.tif'
+                create_json_sidecar(path_folder_sub_id_bids, item_out)
+
+            path_item_out = os.path.join(path_folder_sub_id_bids, item_out)
+            shutil.copy(path_item_in, path_item_out)
+            counter_move = counter_move + 1
```
Comment (mariehbourget) [inline scripts/curate_wakehealth.py:82]: There seems to be a recursion error at the end of the for loop. I have the following error and the script tries to write a `sub-sub17268` folder.
```
sub-17268
Traceback (most recent call last):
  File "scripts/curate_wakehealth.py", line 160, in <module>
    main(args.data)
  File "scripts/curate_wakehealth.py", line 82, in main
    shutil.copy(path_item_in, path_item_out)
  File "/usr/lib/python3.8/shutil.py", line 415, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.8/shutil.py", line 261, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
IsADirectoryError: [Errno 21] Is a directory: '/home/mhbourget/Documents/20210310_wakeheath/test_bidsi/_curated/sub-17268'
```

File: scripts/curate_wakehealth.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+        print("Counter files before move: " + str(counter_file_in))
+        print("Counter moved files: " + str(counter_move))
+
+    sub_list = os.listdir(output_data)
+    sub_list.sort()
+
+    # Write additional files
```
Comment (mariehbourget) [inline scripts/curate_wakehealth.py:90]: I was unable to test the rest of the scripts with tsv/json files because of the error above.

**Review by mariehbourget — COMMENTED** [review]
Thank you @alexfoias!
I tested the script and left a few comments. It seems to work well, except at the end of the for loop. I can't test the tsv/json because of the error.

File: scripts/curate_wakehealth.py (hunk 1 above)
Comment (alexfoias) [reply scripts/curate_wakehealth.py:53]: okay sounds good. I will tweak the script.

File: scripts/curate_wakehealth.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    os.makedirs(output_data)
+
+    subdatasets = os.listdir(root_data)
+    counter_move = 0
+    counter_file_in = 0
+
+    # Loop across subdirectories
+    for ds in subdatasets:
+        path_dataset = os.path.join(root_data, ds)
+        contents_ds = os.listdir(path_dataset)
```
[the remark this answers is not in the document]
Comment (alexfoias) [reply scripts/curate_wakehealth.py:48]: maybe the best strategy is to create a list of the desired folders that we want to convert to BIDS.

[the remark this answers is not in the document]
Comment (mariehbourget) [reply scripts/curate_wakehealth.py:48]: The list for this first "bidsifying" are: `20210308_extracts`, `20210512_extracts` and `20210520_extracts`.

**Comment by alexfoias** [comment]
@mariehbourget I have updated the script. Could you please try it again ?

The important aspect is to save all the subfolders that you want to convert into a main folder and then point to that one.

I have organized the copy of the data from duke like this: 
Screen Shot 2021-06-02 at 11 15 05 AM

**Review by mariehbourget — APPROVED** [review]
Tested on a local copy of a sub-set of folders and images (with the addition of the `-o` flag) It works very well. Everything LGTM , thanks @alexfoias.

Once the dataset is bidsified, we will want to git-annex it (as a private dataset I think).

**Comment by alexfoias** [comment]
@jcohenadad should I push the generated dataset to local git annex server ?

**Comment by mariehbourget** [comment]
Hi @alexfoias, we received new files yesterday from wakehealth and @hermancollin converted them here: `duke/projects/axondeepseg/wakehealth/20210602_extracts/`

Could you add them to the BIDS dataset? Thanks!

**Comment by mariehbourget** [comment]
Hi @alexfoias, let us know when the dataset is ready, do you need any additional information? Thanks!

**Comment by alexfoias** [comment]
@mariehbourget @jcohenadad I have created a repo here: data.neuro.polymtl.ca:datasets/wakehealth

**Comment by alexfoias** [comment]
I used the script from above to generate it.

**Comment by hermancollin** [comment]
Hi @alexfoias. I have uploaded the rest of the dataset to `duke/projects/axondeepseg/wakehealth/20210617_extracts/`. Could you please "bidsify" the data and add to the rest of the repo? I would run the script myself but I don't have writing priviledge on the git-annex repo.

**Comment by jcohenadad** [comment]
> Hi @alexfoias. I have uploaded the rest of the dataset to `duke/projects/axondeepseg/wakehealth/20210617_extracts/`. Could you please "bidsify" the data and add to the rest of the repo? I would run the script myself but I don't have writing priviledge on the git-annex repo.

I think @kousu can grant you writing privilege (you would just need to create a branch)

**Comment by alexfoias** [comment]
@hermancollin Did you manage to push the data ?

**Comment by hermancollin** [comment]
> @hermancollin Did you manage to push the data ?

yes I did (with the help of @kousu). The update was documented in #96. I did have some trouble pushing.

**Pull request merged**